stairway/src/main/java/bio/terra/stairway/impl/FlightFilterAccess.java

```diff
@@ -150,52 +155,22 @@ String makeSql() {
   String makeCountSql() {
     StringBuilder sb = new StringBuilder();
     sb.append("SELECT CURRENT_TIMESTAMP AS currenttime, COUNT(*) AS totalflights FROM ");
-    makeSqlQueryCommon(sb);
+    makeSqlForm(sb);
     return sb.toString();
   }
 
-  // Make the common form of the query string
-  private void makeSqlQueryCommon(StringBuilder sb) {
-    // Decide which form of the query to build.
-    switch (filter.getInputPredicates().size()) {
-      case 0: // Form 1
-        makeSqlForm1(sb);
-        break;
-      case 1: // Form 2
-        makeSqlForm2(sb);
-        break;
-      default: // Form 3
-        makeSqlForm3(sb);
-    }
-  }
+  private void makeSqlForm(StringBuilder sb) {
+    sb.append(FlightDao.FLIGHT_TABLE).append(" F WHERE (1=1)");
 
-  // Form1: only flight table filtering
-  private void makeSqlForm1(StringBuilder sb) {
-    sb.append(FlightDao.FLIGHT_TABLE).append(" F");
-    makeFlightFilter(sb, " WHERE ");
-  }
+    for (FlightFilterPredicate predicate : filter.getInputPredicates()) {
+      sb.append(" AND ").append(makeInputPredicateSql(predicate));
+    }
 
-  // Form2: flight table filtering and a single input parameter filter
-  private void makeSqlForm2(StringBuilder sb) {
-    sb.append(FlightDao.FLIGHT_TABLE)
-        .append(" F INNER JOIN ")
-        .append(FlightDao.FLIGHT_INPUT_TABLE)
-        .append(" I ON F.flightid = I.flightid WHERE ")
-        .append(makeInputPredicateSql(filter.getInputPredicates().get(0)));
-    makeFlightFilter(sb, " AND ");
-  }
+    if (filter.getInputBooleanOperationExpression() != null) {
+      sb.append(" AND ")
+          .append(makeBooleanExpressionsFilters(filter.getInputBooleanOperationExpression()));
+    }
 
-  // Form3: flight table filtering and more than one input parameter filter
-  private void makeSqlForm3(StringBuilder sb) {
-    sb.append(FlightDao.FLIGHT_TABLE)
-        .append(" F INNER JOIN (SELECT flightid, COUNT(*) AS matchCount")
-        .append(" FROM ")
-        .append(FlightDao.FLIGHT_INPUT_TABLE)
-        .append(" I WHERE ");
-    makeInputFilter(sb);
-    sb.append(" GROUP BY I.flightid) INPUT ON F.flightid = INPUT.flightid")
-        .append(" WHERE INPUT.matchCount = ")
-        .append(filter.getInputPredicates().size());
     makeFlightFilter(sb, " AND ");
```

Comment:
I updated the PR so that inputs can be included too.  That was a great call to add.  I've update the unit tests to have examples of those being mixed in too

Reply:
I had thought about that but then you get a weird case with supporting the existing API that returns either input or flight filters.  I went a separate approach and left those but added a top level boolean expression that now takes input and flight (and could potentially take output) filters.  So it's really two ways to construct filtering